src/ApacheOrcDotNet.OptimizedReaderTest.App/ReadAllApp.cs

```diff
@@ -0,0 +1,120 @@
+﻿using ApacheOrcDotNet.OptimizedReader;
+using ApacheOrcDotNet.OptimizedReader.Infrastructure;
+using System;
+using System.Diagnostics;
+using System.Globalization;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace ApacheOrcDotNet.OptimizedReaderTest.App
+{
+    public class ReadAllApp
+    {
+        private readonly string _orcFileUri;
+        private readonly IByteRangeProviderFactory _byteRangeProviderFactory;
+
+        public ReadAllApp(string orcFileUri, IByteRangeProviderFactory byteRangeProviderFactory)
+        {
+            _orcFileUri = orcFileUri;
+            _byteRangeProviderFactory = byteRangeProviderFactory;
+        }
+
+        public async Task Run()
+        {
+            //
+            var watch = new Stopwatch();
+            var configs = new OrcReaderConfiguration();
+            var rangeProvider = _byteRangeProviderFactory.Create(_orcFileUri);
+            var reader = new OptimizedReader.OrcReader(configs, rangeProvider);
+
+            watch.Start();
+
+            // Columns
+            var sourceColumn = reader.GetColumn("source");
```

Comment:
I'd prefer to pull "source", "symbol", "time", and "size" out of here if that's not a lot of work.  You're generating the input files from scratch anyway right?  Just call them "string", "decimal", "int".